lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -133,33 +133,34 @@ sub job_module_stats($) {
     my $result_stat = {};
 
     my $schema = OpenQA::Scheduler::schema();
-    my @ids;
-    while (my $j = $jobs->next) { push(@ids, $j->id); }
-    $jobs->reset;
 
-    for my $id (@ids) {
+    if (ref($jobs) ne 'ARRAY') {
+        my @ids;
+        while (my $j = $jobs->next) { push(@ids, $j->id); }
+        $jobs->reset;
+        $jobs = \@ids;
```

Comment:
actually it's not. $jobs is the parameter and I need to reset the cursor when I iterated over it. The real bug is abusing a parameter for something different :)